chore(docs): modified compiler target code example (#3465)
winglang/wing: docs/docs/02-concepts/03-compile-targets.md

With the Wing `util` library, you can access environment variables. The `WING_TARGET` environment variable contains the current compiler target, which you can use to conditionally run target-specific code. See the example below:

> ```js play

Reviewer: ```suggestion
```js playground
```

bring cloud;
bring util;

> let invokationCounter = new cloud.Counter();

Reviewer: ```suggestion
let invocationCounter = new cloud.Counter();
```

let queue = new cloud.Queue();

queue.setConsumer(inflight (msg: str) => {

>   invokationCounter.inc();

Reviewer: ```suggestion
  invocationCounter.inc();
```

});

new cloud.Function(inflight ()=> { 
  // push a message to queue
  queue.push("m");
  // sleep according to target 
  if util.env("WING_TARGET") == "sim" {
    log("Running on Simulator, sleeping for 1s");
    util.sleep(1s);
  } else {
    log("Running on the cloud, sleeping for 30s");
    util.sleep(30s);
  }

>   log("Function invoked ${invokationCounter.peek()} times");

Author: ```suggestion
  log("Function invoked ${invocationCounter.peek()} times");
```

});
```

> In this example, we're want to sleep briefly for the Simulator target and for 30 seconds for cloud targets, this is done according to `WING_TARGET` environment variable. 

Reviewer: ```suggestion
In this example, we want to sleep briefly for the Simulator target and for 30 seconds for cloud targets, this is achieved using the `WING_TARGET` environment variable. 
```
